src/util/HaltingAnalysis.ts

```diff
@@ -66,46 +109,65 @@ export function haltingAnalysis(
     throw new Error("Halting analysis only supports order 1 at this point");
   }
   const failureCases: HaltingFailure[] = [];
-  const analysisNodes = createAnalysisStructure(nodes);
-
-  const myNode = analysisNodes.find(n => n.networkObject.distance == 0);
-  if (!myNode) {
-    throw new Error("No node with distance 0 in halting analysis");
+  const { root, entries: analysisNodes } = createAnalysisStructure(nodes);
+  function getNode(name: string): AnalysisNode {
+    return analysisNodes.find(n => n.name == name) as AnalysisNode;
   }
-
   // Actual analysis
   // Run through each node and observe the effects of failing it
   analysisNodes.forEach(nodeToHalt => {
-    if (nodeToHalt === myNode) return;
+    if (nodeToHalt === root) return;
 
     reset(analysisNodes);
 
     let deadNodes: NetworkGraphNode[] = [];
 
-    function checkDependents(node: AnalysisGraphNode) {
-      node.dependents.forEach(node => {
-        let threshold = node.networkObject.qset.t;
-        node.dependencies.forEach(dependent => {
-          if (dependent.live) {
+    nodeToHalt.live = false;
+    checkDependents(nodeToHalt);
+    /*
+     * Check all the nodes that are dependent on this newly dead node to see if they go
+     * down as well
+     * @param { AnalysisNode } deadNode - A node that is no longer live
+     */
+    function checkDependents(deadNode: AnalysisNode) {
+      deadNode.dependentsNames.forEach(nodeName => {
+        const dependentNode = getNode(nodeName);
+        // If this node is currently live, but can't make threshold it
+        // goes down, and this error can propagate out.
+        if (dependentNode.live && !checkSubquorum(dependentNode.quorumSet)) {
+          dependentNode.live = false;
+          deadNodes.push(dependentNode.networkObject);
+          checkDependents(dependentNode);
+        }
+      });
+    }
+
+    /*
+     *  Check if this quorum set has enough live nodes to validate
+     *  @param { AnalysisQuorumSet } quorum - Quorum set to test
+     *  @return { boolean } true if this quorum set meets its threshold of valid nodes
+     */
+
+    function checkSubquorum(quorum: AnalysisQuorumSet): boolean {
```

Comment:
Since the `checkDependents` function simply iterates through the dependents but doesn't return anything, I think it might make sense to use a different naming structure for this function that actually returns a boolean. Maybe something like `doesQuorumSetMeetThreshold()` or `quorumSetMeetsThreshold()`?